toolchain/lower/lower.cpp

```diff
@@ -12,18 +12,23 @@
 
 namespace Carbon::Lower {
 
+// TODO: Remove module_name argument.
 auto LowerToLLVM(
     llvm::LLVMContext& llvm_context,
     llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
-    llvm::raw_ostream* llvm_verifier_stream, bool want_debug_info,
     llvm::ArrayRef<Parse::GetTreeAndSubtreesFn> tree_and_subtrees_getters,
-    llvm::StringRef module_name, const SemIR::File& sem_ir,
-    const SemIR::InstNamer* inst_namer, llvm::raw_ostream* vlog_stream)
+    const SemIR::File& sem_ir, const LowerToLLVMOptions& options)
     -> std::unique_ptr<llvm::Module> {
-  Context context(llvm_context, std::move(fs), want_debug_info,
-                  tree_and_subtrees_getters, module_name, vlog_stream);
-  context.GetFileContext(&sem_ir, inst_namer).LowerDefinitions();
-  return std::move(context).Finalize(llvm_verifier_stream);
+  Context context(llvm_context, std::move(fs), options.want_debug_info,
+                  tree_and_subtrees_getters, sem_ir.filename(),
+                  options.vlog_stream);
+
+  // TODO: Consider disabling instruction naming by default if we're not
+  // producing textual LLVM IR.
```

Comment:
I guess a downside of sinking this to here is that we don't know whether we're producing textual IR from `LowerToLLVM`. But that's a problem for later :) We can turn that information into a flag we pass in, if needed.

Reply:
Part of the request I'm trying to address is that this should handle producing textual IR, so it'll know soon. :)